### DISCUSSION TITLE
RuntimeError: find too many unsuccessfully terminated jobs. Too many FP tasks are not converged.

### DISCUSSION THREAD

**Discussion by goodluck-xyz** [deepmodeling/dpgen#898] [Q&A]
When I run dpgen,  there are error in  fp.

The error like this:

INFO:dpgen:failed tasks:      0 in     50    0.00 %
INFO:dpgen:failed frame:     50 in     50   100.00 %
Traceback (most recent call last):
  File "/usr/local/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/dist-packages/dpgen/main.py", line 185, in main
    args.func(args)
  File "/usr/local/lib/python3.8/dist-packages/dpgen/generator/run.py", line 3642, in gen_run
    run_iter (args.PARAM, args.MACHINE)
  File "/usr/local/lib/python3.8/dist-packages/dpgen/generator/run.py", line 3631, in run_iter
    post_fp (ii, jdata)
  File "/usr/local/lib/python3.8/dist-packages/dpgen/generator/run.py", line 3508, in post_fp
    post_fp_vasp(iter_index, jdata)
  File "/usr/local/lib/python3.8/dist-packages/dpgen/generator/run.py", line 3177, in post_fp_vasp
    raise RuntimeError("find too many unsuccessfully terminated jobs. Too many FP tasks are not converged. Please check your input parameters (e.g. INCAR) or configuration (e.g. POSCAR) in directories \'iter.*.*/02.fp/task.*.*/.\'")
RuntimeError: find too many unsuccessfully terminated jobs. Too many FP tasks are not converged. Please check your input parameters (e.g. INCAR) or configuration (e.g. POSCAR) in directories 'iter.*.*/02.fp/task.*.*/.'

I have run the vasp program separately. It run successfully.
Especially, when NSW=10 or 20, the frame failed ratio is 100%. While NSW =1 , the frame failed ratio is 20%. I think the problem is the judgement about failed frame. Excepted the vasp error, what will lead the failed frame.

**Comment by HuangJiameng** [comment]
https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L3177
You can find the function that raised the error here. If OUTCAR is abnormal, the frame will be regarded as failed. Could you check `fp.log` to find out why VASP jobs failed?

**Reply by goodluck-xyz** [reply]
I have cheak the fp.log. The vasp normally finish. Just a warning about the unoccupied bands.
https://github.com/deepmodeling/dpgen/blob/master/dpgen/generator/run.py#L3134
This judgement will classify multi-ionic steps into failed. I think this judgement should be changed into judging whether the last ionic step exists.

**Reply by HuangJiameng** [reply]
Besides, NSW should be set as 1. The INCAR for FP step is prepared for self-consistent field calculation.

**Reply by Guancred** [reply]
whether the NSW can be set as 0? while the similar error occurred when NSW=0
